src/storage/admin/RebuildIndexTask.h

```diff
@@ -36,7 +36,11 @@ class RebuildIndexTask : public AdminTask {
                                                    const IndexItems& items,
                                                    kvstore::RateLimiter* rateLimiter) = 0;
 
-  void cancel() override { canceled_ = true; }
+  void cancel() override {
+    canceled_ = true;
+    auto queueStat = AdminTaskStatus::QUEUE;
+    taskStatus_.compare_exchange_strong(queueStat, AdminTaskStatus::CANCELED);
```

Comment:
What about the RUNNING tasks?

Reply:
subtask will check the canceled_ flag in every loop(loop for rocksdb iterator), and if canceled_ is true, subtask will return, and AdminTaskManager::schedule() will finish and erase this task form taskQueue.

Reply:
🆗